easybuild/framework/easyconfig/tweak.py

```diff
@@ -820,21 +820,29 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
     :return: Location of the modified easyconfig file
     """
     # Fully parse the original easyconfig
-    parsed_ec = process_easyconfig(ec_spec, validate=False)[0]
+    parsed_ec = process_easyconfig(ec_spec, validate=False)[0]['ec']
+
     # Replace the toolchain if the mapping exists
-    tc_name = parsed_ec['ec']['toolchain']['name']
+    tc_name = parsed_ec['toolchain']['name']
     if tc_name in toolchain_mapping:
         new_toolchain = toolchain_mapping[tc_name]
-        _log.debug("Replacing parent toolchain %s with %s", parsed_ec['ec']['toolchain'], new_toolchain)
-        parsed_ec['ec']['toolchain'] = new_toolchain
+        _log.debug("Replacing parent toolchain %s with %s", parsed_ec['toolchain'], new_toolchain)
+        parsed_ec['toolchain'] = new_toolchain
 
     # Replace the toolchains of all the dependencies
     for key in DEPENDENCY_PARAMETERS:
         # loop over a *copy* of dependency dicts (with resolved templates);
-        # to update the original dep dict, we need to index with idx into self._config[key][0]...
-        for idx, dep in enumerate(parsed_ec['ec'][key]):
+        # to update the original dep dict, we need to get a reference with templating disabled...
+        val = parsed_ec[key]
+        orig_val = parsed_ec.get_ref(key)
+
+        if key in parsed_ec.iterate_options:
+            val = flatten(val)
+            orig_val = flatten(orig_val)
```

Comment:
@bartoldeman Same here, `flatten` creates a new list, so this is no longer a reference to the original list...

Reply:
Same here, it's OK to use `flatten` here, since we're still using references to the original `dict` values, and that's what wer'e changing (not the list itself).